File: php/src/Application/Catalogue/ListItems.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

final readonly class ListItems
{
    public function __construct(private ItemRepository $repository)
    {
    }

    /** @return list<ItemView> */
    public function execute(): array
    {
        return $this->repository->findAll();
    }
}

File: java/src/main/java/com/alxarafe/app/application/catalogue/ListItems.java
package com.alxarafe.app.application.catalogue;

import java.util.List;

public final class ListItems {
    private final ItemRepository repository;

    public ListItems(ItemRepository repository) {
        this.repository = repository;
    }

    public List<ItemView> execute() {
        return repository.findAll();
    }
}